docs/digital_analog_qc/daqc-cnot.md

```diff
@@ -43,160 +43,166 @@ $$\begin{aligned}
+Thus, a `CNOT` gate can be created by combining a few single-qubit gates together with a two-qubit Ising interaction between the control and the target qubit which is the essence of the Ising transform proposed in the seminal DAQC paper[^2] for $ZZ$ interactions. In Qadence, both $ZZ$ and $NN$ interactions are supported.
 
-## CNOT in an interacting system of 3 qubits
+## `CNOT` in an interacting system of three qubits
 
-Consider a simple experimental setup with $n=3$ interacting qubits in a triangular grid. For simplicity let's consider that all qubits interact with each other with an Ising ($NN$) interaction of constant strength $g_\text{int}$. The Hamiltonian for the system can be written by summing this interaction over all pairs:
+Consider a simple experimental setup with $n=3$ interacting qubits laid out in a triangular grid. For the sake of simplicity, all qubits interact with each other with an $NN$-Ising interaction of constant strength $g_\text{int}$. The Hamiltonian for the system can be written by summing interaction terms over all pairs:
 
 $$\mathcal{H}_\text{sys}=\sum_{i=0}^{n}\sum_{j=0}^{i-1}g_\text{int}N_iN_j,$$
 
 which in this case leads to only three interaction terms,
 
 $$\mathcal{H}_\text{sys}=g_\text{int}(N_0N_1+N_1N_2+N_0N_2)$$
 
-This generator can be easily built:
-
+This generator can be easily built in Qadence:
 
 ```python exec="on" source="material-block" result="json" session="daqc-cnot"
+from qadence import add, kron
 n_qubits = 3
 
+# Interaction strength.
 g_int = 1.0
 
+# Build a list of interactions.
 interaction_list = []
 for i in range(n_qubits):
     for j in range(i):
-        interaction_list.append(g_int * qd.kron(N(i), N(j)))
+        interaction_list.append(g_int * kron(N(i), N(j)))
 
-h_sys = qd.add(*interaction_list)
+h_sys = add(*interaction_list)
 
-print(h_sys)
+print(f"h_sys = {h_sys}") # markdown-exec: hide
 ```
 
-Now let's consider that the experimental system is fixed, and we cannot isolate the qubits from each other. All we can do is the following:
+Now let's consider that the experimental system is fixed, and qubits can not be isolated one from another. The options are:
 
 - Turn on or off the global system Hamiltonian.
-- Perform single-qubit rotations on individual qubits.
-
-How can we perform a CNOT on two specific qubits of our choice?
-
-To perform a *fully digital* CNOT we would need to isolate the control and target qubit from the third one and have those interact to implement the gate directly. While this may be relatively simple for a 3-qubit system, the experimental burden becomes much greater when we start going into the dozens of qubits.
+- Perform single-qubit rotations.
```

Comment:
```suggestion
- Perform local single-qubit rotations.
```
since it enforces we do require single-qubit addressability, rather than performing the same single-qubit rotation on all qubits.

Reply:
Aren't "local" and "single-qubits" synonymous ?

Reply:
I wouldn't say so. There is a difference between being able to send the same single-qubit pulse to all qubits at the same time or having true local addressing of sending the pulse to a single-qubit. Here we really require the latter.

Reply:
OK, but that would mean that in the former case there would be "global single-qubit pulses" ?

Reply:
Yes indeed. At least that's my understanding of local adressability and global addressability.

Reply:
That would be very confusing terminology IMHO.